From f7c23d3e2171ce6da29fcdf98619087849fd81ee Mon Sep 17 00:00:00 2001
From: Bijan Regmi <vizandabijan@gmail.com>
Date: Sat, 5 Feb 2022 17:42:07 +0545
Subject: [PATCH] CenterFirstWindow

---
 dwm.c        | 5 ++++-
 vanitygaps.c | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index bf979db..ca25190 100644
--- a/dwm.c
+++ b/dwm.c
@@ -109,7 +109,7 @@ struct Client {
 	int basew, baseh, incw, inch, maxw, maxh, minw, minh;
 	int bw, oldbw;
 	unsigned int tags;
-	int isfixed, isfloating, isurgent, neverfocus, oldstate, isfullscreen, issticky;
+	int isfixed, isfloating, isurgent, neverfocus, oldstate, isfullscreen, issticky, CenterThisWindow;
 	Client *next;
 	Client *snext;
 	Monitor *mon;
@@ -167,6 +167,7 @@ typedef struct {
 	const char *title;
 	unsigned int tags;
 	int isfloating;
+	int CenterThisWindow;
 	int monitor;
 } Rule;
 
@@ -349,6 +350,7 @@ applyrules(Client *c)
 
 	/* rule matching */
 	c->isfloating = 0;
+        c->CenterThisWindow = 0;
 	c->tags = 0;
 	XGetClassHint(dpy, c->win, &ch);
 	class    = ch.res_class ? ch.res_class : broken;
@@ -361,6 +363,7 @@ applyrules(Client *c)
 		&& (!r->instance || strstr(instance, r->instance)))
 		{
 			c->isfloating = r->isfloating;
+			c->CenterThisWindow = r->CenterThisWindow;
 			c->tags |= r->tags;
 			for (m = mons; m && m->num != r->monitor; m = m->next);
 			if (m)
diff --git a/vanitygaps.c b/vanitygaps.c
index 1a816b6..97591b3 100644
--- a/vanitygaps.c
+++ b/vanitygaps.c
@@ -819,4 +819,11 @@ tile(Monitor *m)
 			resize(c, sx, sy, sw - (2*c->bw), sh * (c->cfact / sfacts) + ((i - m->nmaster) < srest ? 1 : 0) - (2*c->bw), 0);
 			sy += HEIGHT(c) + ih;
 		}
+	
+	if (n == 1 && selmon->sel->CenterThisWindow)
+        resizeclient(selmon->sel,
+                (selmon->mw - selmon->mw * 0.66) / 2,
+                (selmon->mh - selmon->mh * 0.75) / 2,
+                selmon->mw * 0.66,
+                selmon->mh * 0.75);
 }
\ No newline at end of file
-- 
2.35.1

