From 687a4fa56e948558b6bb5dc108c7655b61a9b746 Mon Sep 17 00:00:00 2001
From: Bijan Regmi <vizandabijan@gmail.com>
Date: Fri, 19 Nov 2021 19:58:53 +0545
Subject: [PATCH] theme

---
 config.def.h | 55 +++++++++++++++++++++++++---------------------------
 1 file changed, 26 insertions(+), 29 deletions(-)

diff --git a/config.def.h b/config.def.h
index 3d86161..3074800 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "JetBrainsMono Nerd Font:style=Regular:pixelsize=12:antialias=true:autohint=true";
 static int borderpx = 2;
 
 /*
@@ -98,31 +98,28 @@ float alpha = 0.5;
 
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
-	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
-
-	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
-
-	[255] = 0,
-
-	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
+  	[0] = "#000000", /* black   */
+  	[1] = "#ff5555", /* red     */
+  	[2] = "#50fa7b", /* green   */
+  	[3] = "#f1fa8c", /* yellow  */
+  	[4] = "#bd93f9", /* blue    */
+  	[5] = "#ff79c6", /* magenta */
+  	[6] = "#8be9fd", /* cyan    */
+  	[7] = "#bbbbbb", /* white   */
+
+  	/* 8 bright colors */
+  	[8]  = "#44475a", /* black   */
+  	[9]  = "#ff5555", /* red     */
+  	[10] = "#50fa7b", /* green   */
+  	[11] = "#f1fa8c", /* yellow  */
+  	[12] = "#bd93f9", /* blue    */
+  	[13] = "#ff79c6", /* magenta */
+  	[14] = "#8be9fd", /* cyan    */
+  	[15] = "#ffffff", /* white   */
+
+  	/* special colors */
+  	[256] = "#282a36", /* background */
+  	[257] = "#f8f8f2", /* foreground */
 };
 
 
@@ -130,9 +127,9 @@ static const char *colorname[] = {
  * Default colors (colorname index)
  * foreground, background, cursor, reverse cursor
  */
-unsigned int defaultfg = 7;
+unsigned int defaultfg = 257;
 unsigned int defaultbg = 0;
-static unsigned int defaultcs = 256;
+static unsigned int defaultcs = 257;
 static unsigned int defaultrcs = 257;
 
 /*
@@ -142,7 +139,7 @@ static unsigned int defaultrcs = 257;
  * 6: Bar ("|")
  * 7: Snowman ("☃")
  */
-static unsigned int cursorshape = 2;
+static unsigned int cursorshape = 6;
 
 /*
  * Default columns and rows numbers
-- 
2.33.1

